ironfish/src/blockchain/blockchain.ts

```diff
@@ -1418,7 +1418,7 @@ export class Blockchain {
     transaction: Transaction,
     tx: IDatabaseTransaction,
   ): Promise<void> {
-    for (const { asset, value } of transaction.mints.reverse()) {
+    for (const { asset, value } of transaction.mints.slice().reverse()) {
```

Comment:
Might be worth turning some of these into readonly arrays, like

`public readonly mints: readonly MintDescription[]`

Reply:
Yeah, I guess the entire Transaction should be mutable.